pkg/clients/helm/client.go

```diff
@@ -221,9 +221,11 @@ func (hc *client) pullChart(spec *v1beta1.ChartSpec, creds *RepoCreds, chartDir
 
 	pc.DestDir = chartDir
 
-	err := hc.login(spec, creds, pc.InsecureSkipTLSverify)
-	if err != nil {
-		return err
+	if pc.Username != "" || pc.Password != "" {
```

Comment:
nit: It may be more intentional to check `creds.Username` and `creds.Password` here, instead of the `pc` fields, because the `creds` object is what gets passed into `hc.login` to actually be used for the login operation as seen on https://github.com/crossplane-contrib/provider-helm/blob/master/pkg/clients/helm/client.go#L250.  So while this check is functionally equivalent due to the previous assignment on lines 219-220, it may just be cleaner and more intentional to check the variables that will actually be used :)

Reply:
Agreed 👍🏼